packages/tasit-action/src/NFT.test.js

```diff
@@ -189,5 +205,103 @@ describe("TasitAction.NFT", () => {
+      expect(senderBalance.toNumber()).to.equal(1);
+
+      receiverBalance = await fullNFT.balanceOf(bob.address);
+      expect(receiverBalance.toNumber()).to.equal(0);
+
+      txSubscription = fullNFT.safeTransferFrom(
+        ana.address,
+        bob.address,
+        tokenId
+      );
+
+      await txSubscription.waitForNonceToUpdate();
+
+      senderBalance = await fullNFT.balanceOf(ana.address);
+      expect(senderBalance.toNumber()).to.equal(0);
+
+      receiverBalance = await fullNFT.balanceOf(bob.address);
+      expect(receiverBalance.toNumber()).to.equal(1);
+    });
+
+    it("shouldn't do a safeTransferFrom to a contract that doesn't implement `onERC721Received` - Contract Subscription", async () => {
+      let senderBalance, receiverBalance;
+      const sender = ana.address;
+      const receiver = simpleStorageWithRemovedAddress;
+      const contractErrorFakeFn = sinon.fake();
+
+      fullNFT = new NFT(fullNFTAddress, ana);
+
+      senderBalance = await fullNFT.balanceOf(sender);
+      expect(senderBalance.toNumber()).to.equal(1);
+
+      receiverBalance = await fullNFT.balanceOf(receiver);
+      expect(receiverBalance.toNumber()).to.equal(0);
+
+      const contractErrorListener = message => {
+        const { error, eventName } = message;
+        expect(error.message).to.equal(
+          "Action with error: VM Exception while processing transaction: revert"
+        );
+        contractErrorFakeFn();
+      };
+
+      fullNFT.on("error", contractErrorListener);
+
+      txSubscription = fullNFT.safeTransferFrom(sender, receiver, tokenId);
+      await txSubscription.waitForNonceToUpdate();
+
+      expect(contractErrorFakeFn.called).to.be.true;
+
+      senderBalance = await fullNFT.balanceOf(sender);
+      expect(senderBalance.toNumber()).to.equal(1);
+
+      receiverBalance = await fullNFT.balanceOf(receiver);
+      expect(receiverBalance.toNumber()).to.equal(0);
+    });
+
+    it("shouldn't do a safeTransferFrom to a contract that doesn't implement `onERC721Received` - Action Subscription", async () => {
+      let senderBalance, receiverBalance;
+      const sender = ana.address;
+      const receiver = simpleStorageWithRemovedAddress;
+      const actionErrorFakeFn = sinon.fake();
+
+      fullNFT = new NFT(fullNFTAddress, ana);
+
+      senderBalance = await fullNFT.balanceOf(sender);
+      expect(senderBalance.toNumber()).to.equal(1);
+
+      receiverBalance = await fullNFT.balanceOf(receiver);
+      expect(receiverBalance.toNumber()).to.equal(0);
+
+      const actionErrorListener = message => {
+        const { error, eventName } = message;
+        expect(error.message).to.equal(
+          "Action with error: VM Exception while processing transaction: revert"
+        );
+        actionErrorFakeFn();
+      };
+
+      txSubscription = fullNFT.safeTransferFrom(sender, receiver, tokenId);
+      txSubscription.on("error", actionErrorListener);
```

Comment:
I haven't yet thought of a good way to do this, but it would be cool if we could instantiate the `txSubscription` object and attach the error listener before sending the transaction somehow. But I don't have a concrete suggestion of how to do that in a way that would be a more understandable improvement for the end user.

Reply:
Brainstorming:
```javascript
// 1
txSubscription = fullNFT.safeTransferFrom(sender, receiver, tokenId).on("error", listener);

// 2
action = fullNFT.safeTransferFrom(sender, receiver, tokenId); 
action.on("error", listener);
acton.send();
```

Reply:
Oooooh I love 2!

That way it naturally fits with the meta-tx-first mindset, that a transaction might not necessarily be paid for and sent right away by the user.

Reply:
And that way, calling write functions on a contract class just build the raw tx object and return it

Reply:
New issue?

Reply:
Sounds good! I'll let you create it. Thanks. Feel free to resolve this once you do.

Reply:
https://github.com/tasitlabs/TasitSDK/issues/162